citron-collection-partitioner/collection_partitioner.py

```diff
@@ -0,0 +1,160 @@
+import pandas as pd
+import logging as log
+import re
+
+import boto3
+from botocore.exceptions import NoCredentialsError
+
+
+class CollectionPartitioner:
+    def __init__(self, bucket: str):
+        self.bucket = bucket
+
+    @staticmethod
+    def init_sts_client():
+        """Init client for AWS S3"""
+        try:
+            s3_client = boto3.client("sts")
+        except NoCredentialsError:
+            log.error("Credentials are missing or incorrect")
+            raise
+        return s3_client
+
+    def get_files_of_collection(self, collection_key: str) -> list:
+        """
+
+        :param collection_key: key of the collection folder
+        :return: list of keys of all the files in the collection folder
+        """
+
+        keys = []
+
+        sts_client= CollectionPartitioner.init_sts_client()
+        assumed_role_object = sts_client.assume_role(
+            RoleArn="arn:aws:iam::585762237892:role/gads-citron-modeo",
```

Comment:
This STRING should be defined as a global variable at the top of the file : 

`CITRON_ROLE_ARN = arn:aws:iam::585762237892:role/gads-citron-modeo`